Configurations/SDKAdditions.xcconfig

```diff
@@ -26,10 +26,11 @@
 // FIXME: Add unversioned sparse SDKs for headers and framework stubs that we can share between OS versions.
 //WK_HAS_UNVERSIONED_SDK_ADDITIONS = ;
 
-WK_HAS_VERSIONED_SDK_ADDITIONS = YES;
-WK_HAS_VERSIONED_SDK_ADDITIONS[sdk=macos*] = $(WK_HAS_VERSIONED_SDK_ADDITIONS_$(USE_INTERNAL_SDK);
-WK_HAS_VERSIONED_SDK_ADDITIONS_ = $(WK_HAS_VERSIONED_SDK_ADDITIONS$(WK_MACOS_1400));
-WK_HAS_VERSIONED_SDK_ADDITIONS_MACOS_SINCE_1400 = YES;
+WK_HAS_VERSIONED_SDK_ADDITIONS = $(WK_NOT_$(USE_INTERNAL_SDK));
+// Prior to Sonoma, downlevel macOS builds link against AppleJPEGXL.framework.
```

Comment:
Nitpick: STP too.

Reply:
Less of a nitpick: seems like we will take all open source SDK additions for internal builds, not just the necessary AppleJPEGXL?

Reply:
We should only take AppleJPEGXL, because that's the only thing in e.g. `macosx13.0.internal-additions.sdk`. The logic to choose between internal and public SDK additions landed in #15464, which I realize makes it difficult to see what's happening from this PR. But in WebKitProjectPaths.xcconfig:

```
WK_VERSIONED_SDK_ADDITIONS_PATH = $(WK_VERSIONED_SDK_ADDITIONS_PATH_$(USE_INTERNAL_SDK));
WK_VERSIONED_SDK_ADDITIONS_PATH_ = $(WK_WORKSPACE_DIR)/WebKitLibraries/SDKs/$(PLATFORM_NAME)$($(DEPLOYMENT_TARGET_SETTING_NAME):base).0-additions.sdk;
WK_VERSIONED_SDK_ADDITIONS_PATH_YES = $(WK_WORKSPACE_DIR)/WebKitLibraries/SDKs/$(PLATFORM_NAME)$($(DEPLOYMENT_TARGET_SETTING_NAME):base).0.internal-additions.sdk;
```